### PR TITLE
[#1658] feat(common): redact sensitive string in catalog properties

### PR DESCRIPTION
**Pull request by zhoukangcn**
### What changes were proposed in this pull request?

redact sensitive string in catalog properties

### Why are the changes needed?

Fix: #1658

### Does this PR introduce _any_ user-facing change?

get catalog proerties from HTTP Rest API will get a redaction string when key reglike '(?i)secret|password|token'

### How was this patch tested?

UT & UI

### CODE REVIEW

File: common/src/main/java/com/datastrato/gravitino/dto/CatalogDTO.java [hunk 1]
```diff
@@ -234,6 +235,11 @@ public CatalogDTO build() {
           StringUtils.isNotBlank(provider), "provider cannot be null or empty");
       Preconditions.checkArgument(audit != null, "audit cannot be null");
 
+      // redact properties
+      if (properties != null) {
```
Comment (yuqi1129) [inline common/src/main/java/com/datastrato/gravitino/dto/CatalogDTO.java:239]: @jerryshao 
I'm not very sure about the point:  Should we only need to apply the action for catalog, what's about `schema` and `table`, possibly there would be sensitive messages in the schema/table properties.

File: common/src/main/java/com/datastrato/gravitino/utils/MapUtils.java [hunk 2]
```diff
@@ -8,9 +8,13 @@
 import com.google.common.collect.Maps;
 import java.util.Collections;
 import java.util.Map;
+import java.util.regex.Pattern;
 
 /** Utility class for working with maps. */
 public class MapUtils {
+  private static final String PROPERTIES_REDACTION_PATTERN = "(?i)secret|password|token";
+  private static final Pattern PROPERTIES_REDACTION = Pattern.compile(PROPERTIES_REDACTION_PATTERN);
+  private static final String REDACTION_REPLACEMENT_TEXT = "*********(redacted)";
```
[the remark this answers is not in the document]
Comment (zhoukangcn) [reply common/src/main/java/com/datastrato/gravitino/utils/MapUtils.java:17]: This references Spark's implementation. I think adding a 'redacted' suffix could make it easier for the user to understand.

File: common/src/main/java/com/datastrato/gravitino/dto/CatalogDTO.java (hunk 1 above)
Comment (zhoukangcn) [reply common/src/main/java/com/datastrato/gravitino/dto/CatalogDTO.java:239]: The most common encryption scenario at present is the connection information for the catalog (mainly password). As for schemas/tables, there is no need for this in our production enviorment

File: common/src/test/java/com/datastrato/gravitino/utils/TestMapUtils.java [hunk 3]
```diff
@@ -24,4 +24,26 @@ public void testGetPrefixMap() {
...
+    Map<String, String> source =
+        ImmutableMap.of(
+            "secret", "value",
+            "password", "value",
+            "token", "value",
+            "jdbc-password", "value",
+            "regular", "value");
+    Map<String, String> expected =
+        ImmutableMap.of(
+            "secret", "*********(redacted)",
```
[the remark this answers is not in the document]
Comment (zhoukangcn) [reply common/src/test/java/com/datastrato/gravitino/utils/TestMapUtils.java]: done

**Comment by mchades** [comment]
When defining catalog property metadata, we do support specifying the ["hidden"](https://github.com/datastrato/gravitino/blob/3b7e01af5c86de45c70b7f252b427799684343ec/core/src/main/java/com/datastrato/gravitino/catalog/PropertyEntry.java#L52) attribute for PropertyEntry. Why don't we directly define these properties as "hidden"? This way, we can automatically hide these properties in the response.

**Pull request closed without merging**